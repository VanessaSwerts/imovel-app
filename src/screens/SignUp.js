import React, { useState } from 'react'
import { Platform, KeyboardAvoidingView, StyleSheet, Text, View, Image, ScrollView, TouchableOpacity } from 'react-native'
import { InputArea, ImagePickerFunction } from '../components'
import { ProgressSteps, ProgressStep } from 'react-native-progress-steps'
import { useNavigation } from '@react-navigation/native'
import { FontAwesome5 } from '@expo/vector-icons'
import api, { STORAGE_URL } from '../services/api'
import { formatPhoneNumber, formatCPF } from '../utils/util'

import colors from '../constants/colors.json'

export default function SignUp() {
  const navigation = useNavigation()

  const [data, setData] = useState({
    name: "",
    email: "",
    confirmEmail: "",
    cpf: "",
    phone: "",
    password: "",
    confirmPassword: "",
    avatar: null
  })

  const onChange = (type, value) => setData({ ...data, [type]: value })

  const signUp = () => {
    const config = {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    }

    const formData = new FormData()

    formData.append('email', data.email)
    formData.append('password', data.password)
    formData.append('name', data.name)
    formData.append('cpf', data.cpf)
    formData.append('phone', data.phone)
    if (data.avatar !== null)
      formData.append('file', data.avatar)

    api.post("/user", formData, config)
      .then((res) => {
        navigation.navigate("SignIn")
      })
      .catch((err) => {
        console.error(err)
      })
  }

  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior="padding"
      enabled={Platform.OS === 'ios'}
    >
      <ScrollView
        alwaysBounceVertical={false}
        showsVerticalScrollIndicator={false}
      >

        <TouchableOpacity style={styles.backButton} onPress={() => navigation.navigate("SignIn")}>
          <FontAwesome5 name={'arrow-left'} size={20} color={colors['yellow']} />
        </TouchableOpacity>


        <View style={styles.header} >
          <Text style={styles.message}> Cadastre-se.</Text>
          <Text style={styles.message}> É rápido, simples e gratuito!</Text>
        </View>

        <ProgressSteps
          activeStepIconBorderColor={colors['yellow']}
          completedProgressBarColor={colors['yellow']}
          activeLabelColor={colors['yellow']}
          completedLabelColor={colors['yellow']}
          completedStepIconColor={colors['yellow']}
          completedCheckColor={colors['platinum']}
        >
          <ProgressStep
            nextBtnText={'Próximo'}
            label="Pessoal"
            nextBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            scrollable={false}
          >
            <View style={styles.containerInput}>
              <Text style={styles.label}>Nome</Text>
              <InputArea
                placeholder={'Ex: Pedro Henrique Santos'}
                value={data.name}
                onChangeText={(value) => onChange("name", value)}
              />
              <Text style={styles.label}>CPF</Text>
              <InputArea
                placeholder={'xxx.xxx.xxx-xx'}
                value={formatCPF(data.cpf)}
                onChangeText={(value) => onChange("cpf", value)}
                keyboardType={'numeric'}

              />
              <Text style={styles.label}>Celular</Text>
              <InputArea
                placeholder={'(xx)xxxxx-xxxx'}
                value={formatPhoneNumber(data.phone)}
                onChangeText={(value) => { onChange("phone", value) }}
                keyboardType={'phone-pad'}
              />
            </View>
          </ProgressStep>

          <ProgressStep
            label="Login"
            nextBtnText={'Próximo'}
            previousBtnText={'Anterior'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
          >
            <View style={styles.containerInput}>
              <Text style={styles.messageEmail}>Por favor, entre com um e-mail válido!</Text>
              <Text style={styles.label}>E-mail</Text>
              <InputArea
                placeholder={'Ex: pedrohs@gmail.com'}
                keyboardType={'email-address'}
                value={data.email}
                onChangeText={(value) => onChange("email", value)}
              />
              <Text style={styles.label}>Confirmar e-mail</Text>
              <InputArea
                placeholder={'Ex: pedrohs@gmail.com'}
                keyboardType={'email-address'}
                value={data.confirmEmail}
                onChangeText={(value) => onChange("confirmEmail", value)}
              />
            </View>
          </ProgressStep>

          <ProgressStep
            label="Senha"
            nextBtnText={'Próximo'}
            previousBtnText={'Anterior'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
          >
            <View style={styles.containerInput}>
              <Text style={styles.messageEmail}>Para sua segurança, a senha deve ter no mínimo 8 caracteres, com números, letra maiúscula e minúscula e caracteres especiais.</Text>
              <Text style={styles.label}>Senha</Text>
              <InputArea
                placeholder={'••••••••'}
                secureTextEntry={true}
                value={data.password}
                onChangeText={(value) => onChange("password", value)}
              />
              <Text style={styles.label}>Confirmar senha</Text>
              <InputArea
                placeholder={'••••••••'}
                secureTextEntry={true}
                value={data.confirmPassword}
                onChangeText={(value) => onChange("confirmPassword", value)}
              />
            </View>
          </ProgressStep>

          <ProgressStep
            label="Foto"
            previousBtnText={'Anterior'}
            finishBtnText={'Finalizar'}
            nextBtnStyle={styles.button}
            previousBtnStyle={styles.button}
            nextBtnTextStyle={styles.buttonText}
            previousBtnTextStyle={styles.buttonText}
            scrollable={false}
            onSubmit={() => signUp()}
          >
            <View style={styles.containerInput}>
              <ImagePickerFunction onChange={(image) => onChange("avatar", image)}>
                <Image
                  source={{ uri: data.avatar ? data.avatar.uri : `${STORAGE_URL}/user/default-avatar.png` }}
                  style={styles.avatar}
                />
              </ImagePickerFunction>
              <ImagePickerFunction onChange={(image) => onChange("avatar", image)}>
                <Text style={styles.avatarText}>Alterar</Text>
              </ImagePickerFunction>
            </View>
          </ProgressStep>

        </ProgressSteps>

      </ScrollView>
    </KeyboardAvoidingView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors["platinum"],
    justifyContent: 'center',
    alignItems: 'center',
    padding: 30,
  },

  containerInput: {
    marginBottom: 20,
  },

  backButton: {
    marginTop: 5,
  },

  header: {
    marginTop: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },

  message: {
    width: 300,
    fontSize: 23,
    fontWeight: "600",
    color: colors['blue'],
    textAlign: 'left',
  },

  messageEmail: {
    marginTop: 10,
    fontSize: 15,
    fontWeight: "500",
    color: '#999',
    textAlign: 'center',

  },

  label: {
    marginTop: 15,
    marginBottom: -10,
    marginLeft: 20,
    fontSize: 14,
    fontWeight: "500",
    color: colors['blue'],
  },

  button: {
    height: 35,
    width: 90,
    backgroundColor: colors['yellow'],
    borderRadius: 24,
    alignItems: 'center',
    justifyContent: 'center',
  },

  buttonText: {
    color: colors['blue'],
    fontWeight: 'bold',
    fontSize: 16,
  },

  avatar: {
    width: "100%",
    height: "100%",
    width: 180,
    height: 180,
    borderRadius: 180,
    borderWidth: 3,
    borderColor: colors["blue"],
    alignSelf: 'center',
  },

  avatarText: {
    color: colors['yellow'],
    fontWeight: 'bold',
    fontSize: 16,
    marginVertical: 5,
    textDecorationLine: "underline",
    alignSelf: 'center',
  }

})