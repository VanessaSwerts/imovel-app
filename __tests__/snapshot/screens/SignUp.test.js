import React from 'react'
import { create } from 'react-test-renderer'
import SignUp from '../../../src/screens/SignUp'

jest.runAllTimers()
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn()
    })
  }
})
jest.mock('../../../src/contexts/loading', () => {
  return {
    ...jest.requireActual('../../../src/contexts/loading'),
    useLoading: () => ({
      startLoading: jest.fn(),
      stopLoading: jest.fn(),
      loading: jest.fn()
    })
  }
})

describe('SignUp snapshot test', () => {
  it('render SignUp screen correctly', () => {
    const tree = create(<SignUp />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
