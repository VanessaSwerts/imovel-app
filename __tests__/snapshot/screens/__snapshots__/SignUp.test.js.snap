// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp snapshot test render SignUp screen correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E5E5E5",
        "flex": 1,
        "justifyContent": "center",
        "padding": 30,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    alwaysBounceVertical={false}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 5,
            "opacity": 1,
          }
        }
      >
        <Icon
          color="#FCA311"
          name="arrow-left"
          size={20}
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#14213D",
              "fontSize": 23,
              "fontWeight": "600",
              "textAlign": "left",
              "width": 300,
            }
          }
        >
           Cadastre-se.
        </Text>
        <Text
          style={
            Object {
              "color": "#14213D",
              "fontSize": 23,
              "fontWeight": "600",
              "textAlign": "left",
              "width": 300,
            }
          }
        >
           É rápido, simples e gratuito!
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginBottom": 50,
              "position": "relative",
              "top": 30,
            }
          }
        >
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#FCA311",
                      "borderRadius": 20,
                      "borderWidth": 5,
                      "bottom": 2,
                      "height": 40,
                      "width": 40,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 6.666666666666667,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "black",
                        }
                      }
                    >
                      1
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "#FCA311",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                >
                  Pessoal
                </Text>
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 48,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18.018018018018015,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      2
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                >
                  Login
                </Text>
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 44,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      3
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                >
                  Senha
                </Text>
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 44,
                      "marginLeft": 22,
                      "position": "absolute",
                      "right": 0,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ebebe4",
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "top": 9,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                        }
                      }
                    >
                      4
                    </Text>
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "color": "lightgray",
                      "flexWrap": "wrap",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "marginTop": 4,
                      "paddingTop": 4,
                      "textAlign": "center",
                      "width": 100,
                    }
                  }
                >
                  Foto
                </Text>
                <View
                  style={
                    Object {
                      "borderTopColor": "#ebebe4",
                      "borderTopStyle": "solid",
                      "borderTopWidth": 3,
                      "left": 0,
                      "marginRight": 22,
                      "position": "absolute",
                      "right": 44,
                      "top": 18,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#14213D",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "marginBottom": -10,
                      "marginLeft": 20,
                      "marginTop": 15,
                    }
                  }
                >
                  Nome
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#FFF",
                        "borderColor": "#14213D",
                        "borderRadius": 25,
                        "borderWidth": 1,
                        "elevation": 2,
                        "flexDirection": "row",
                        "marginTop": 15,
                        "paddingHorizontal": 12,
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 2,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Ex: Pedro Henrique Santos"
                    placeholderTextColor="#d2d2d2"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Object {
                          "height": 46,
                          "paddingHorizontal": 8,
                          "paddingVertical": 10,
                          "width": "100%",
                        },
                        undefined,
                      ]
                    }
                    testID="signUp-name-input"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#14213D",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "marginBottom": -10,
                      "marginLeft": 20,
                      "marginTop": 15,
                    }
                  }
                >
                  CPF
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#FFF",
                        "borderColor": "#14213D",
                        "borderRadius": 25,
                        "borderWidth": 1,
                        "elevation": 2,
                        "flexDirection": "row",
                        "marginTop": 15,
                        "paddingHorizontal": 12,
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 2,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="numeric"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="xxx.xxx.xxx-xx"
                    placeholderTextColor="#d2d2d2"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Object {
                          "height": 46,
                          "paddingHorizontal": 8,
                          "paddingVertical": 10,
                          "width": "100%",
                        },
                        undefined,
                      ]
                    }
                    testID="signUp-cpf-input"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#14213D",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "marginBottom": -10,
                      "marginLeft": 20,
                      "marginTop": 15,
                    }
                  }
                >
                  Celular
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "backgroundColor": "#FFF",
                        "borderColor": "#14213D",
                        "borderRadius": 25,
                        "borderWidth": 1,
                        "elevation": 2,
                        "flexDirection": "row",
                        "marginTop": 15,
                        "paddingHorizontal": 12,
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0.2,
                        "shadowRadius": 2,
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="phone-pad"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="(xx)xxxxx-xxxx"
                    placeholderTextColor="#d2d2d2"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        Object {
                          "height": 46,
                          "paddingHorizontal": 8,
                          "paddingVertical": 10,
                          "width": "100%",
                        },
                        undefined,
                      ]
                    }
                    testID="signUp-phone-input"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginTop": 90,
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 40,
                    "left": 60,
                    "position": "absolute",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "padding": 8,
                      "textAlign": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "bottom": 40,
                    "position": "absolute",
                    "right": 60,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FCA311",
                      "borderRadius": 24,
                      "height": 35,
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 8,
                      "textAlign": "center",
                      "width": 90,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#14213D",
                          "fontSize": 16,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Próximo
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
